apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -844,13 +844,235 @@ def _ingest_obs_var_m_p(
+    measurement_name: str,
+    X_layer_name: str,
+    obs_ids: Sequence[str],
+    var_ids: Sequence[str],
+    *,
+    registration_mapping: ExperimentAmbientLabelMapping,
+    X_kind: type[SparseNDArray] | type[DenseNDArray] = SparseNDArray,
+    context: SOMATileDBContext | None = None,
+    platform_config: PlatformConfig | None = None,
+) -> str:
+    """Appends new data to an existing ``X`` matrix. Nominally to be used in conjunction
+    with ``update_obs`` and ``update_var``, as an itemized alternative to doing
+    ``from_anndata`` with a registration mapping supplied.
+
+    This function is deprecated and will be removed in a future version of this package.
+
+    It is recommended to use ``tiledbsoma.io.from_anndata`` (with a registration map from
+    ``tiledbsoma.io.register_anndatas`` or ``tiledbsoma.io.register_h5ads``) for appending new,
+    complete AnnData objects to an :class:`Experiment`.
+
+    Example::
+
+        rd = tiledbsoma.io.register_anndatas(
+            exp_uri,
+            [new_anndata],
+            measurement_name="RNA",
+            obs_field_name="obs_id",
+            var_field_name="var_id",
+        )
+
+
+        with tiledbsoma.Experiment.open(exp_uri) as exp:
+            tiledbsoma.io.append_X(
+                exp,
+                new_X=adata.X,
+                measurement_name=measurement_name,
+                X_layer_name=X_layer_name,
+                obs_ids=list(new_anndata.obs.index),
+                var_ids=list(new_anndata.var.index),
+                registration_mapping=rd,
+            )
+
+    Lifecycle:
+        Deprecated.
+    """
+    exp.verify_open_for_writing()
+    if measurement_name not in exp.ms:
+        raise SOMAError(f"Experiment {exp.uri} has no measurement named {measurement_name}")
+    if X_layer_name not in exp.ms[measurement_name].X:
+        raise SOMAError(f"Experiment {exp.uri} has no X layer named {X_layer_name} in measurement {measurement_name}")
+    X = exp.ms[measurement_name].X[X_layer_name]
+
+    # Map the user-level ingest mode to a set of implementation-level boolean flags.
+    # See comments in from_anndata.
+    ingestion_params = IngestionParams("write", registration_mapping)
+    context = _validate_soma_tiledb_context(context)
+
+    s = _util.get_start_stamp()
+    _logging.log_io_same(f"Start  writing var for {X.uri}")
+
+    axis_0_mapping = registration_mapping.obs_axis.id_mapping_from_values(obs_ids)
+    axis_1_mapping = registration_mapping.var_axes[measurement_name].id_mapping_from_values(var_ids)
+
+    with _create_from_matrix(
+        X_kind,
+        X.uri,
+        new_X,
+        ingestion_params=ingestion_params,
+        platform_config=platform_config,
+        context=context,
+        axis_0_mapping=axis_0_mapping,
+        axis_1_mapping=axis_1_mapping,
+        must_exist=True,
+    ):
+        _logging.log_io_same(_util.format_elapsed(s, f"Finish writing X for {X.uri}"))
+    return X.uri
+
+
```

Comment:
```suggestion
```
Looks like the rebase accidentally added back in these removed functions. They should be re-removed before merging.